src/base/bittorrent/session.cpp

```diff
@@ -2368,6 +2368,36 @@ void Session::setIPFilterFile(QString path)
     }
 }
 
+void Session::setBannedIPs(const QStringList &newList)
+{
+    if (newList == m_bannedIPs)
```

Comment:
Due to this we must ensure that the lists are ALWAYS sorted.
I would suggest to do this in the widget for `newList`, but his may come from the web api too.
So `newList` must be sorted here first.
`m_bannedIPs` must be sorted during loading. The type of `m_bannedIPs` is of `CachedSettingValue<>`. This template takes a lamda in the constructor as an argument to manipulate the loaded data. If you can't figure out what to do sort it, then leave a message and I'll help you. (or some other one).